blockchain/proxy.go

```diff
@@ -74,13 +84,19 @@ func NewProxy(args ArgsProxy) (*proxy, error) {
 		return nil, err
 	}
 
+	cacher := args.FilterQueryBlockCacher
+	if cacher == nil {
```

Comment:
remove this and let the dev specifically add the DisbledBlockDataCache

Reply:
I do not agree: https://github.com/multiversx/mx-sdk-go/pull/186#discussion_r1753979237